### DISCUSSION TITLE
how to stop microphone during recording when someone speaks above the certain level in wavesurfer.js record module?

### DISCUSSION THREAD

**Discussion by shahbazg4045** [katspaugh/wavesurfer.js#3397] [Q&A]
When the user start the microphone recording and speak a word then is there any event or methods in wavesurfer.js to detect the audio at certain level or above the silence bar then stop immediately?

**Comment by katspaugh** [comment]
When you start listening to the microphone input (by calling startMic), you get a media stream which can be analyzed for loudness/silence.

See https://wavesurfer.xyz/docs/classes/plugins_record.default#startMic

Here’s an example of how to detect silence: https://stackoverflow.com/a/71205954
